### PR TITLE
Replace deprecated getter/setter methods with `Object.defineProperty`

### PR DESCRIPTION
**Pull request by XhmikosR**
Not 100% sure about these, needs more eyes :)

### CODE REVIEW

File: lib/svg-sprite/mode/standalone.js [hunk 1]
```diff
@@ -89,13 +89,20 @@ SVGSpriteStandalone.prototype._layout = function(files, cb, extend) {
...
             Object.defineProperty(this.data.shapes[index], '_svg', {
                 enumerable: false,
                 writable: true
             });
 
-            this.data.shapes[index].__defineSetter__('svg', function(svg) {
-                this._svg = svg;
+            // Create the SVG getter/setter
+            Object.defineProperty(this.data.shapes[index], 'svg', {
+                get() {
```
Comment (XhmikosR) [inline lib/svg-sprite/mode/standalone.js:100]: This might be redundant

**Comment by Kreeg** [comment]
LGTM, but there is more cases in `symbol.js` and `stack.js`

**Comment by XhmikosR** [comment]
Feel free to update the other instances too, IIRC I has some trouble with them (the patch is old, I just rebased it).

File: lib/svg-sprite/mode/standalone.js [hunk 2]
```diff
@@ -89,13 +89,20 @@ SVGSpriteStandalone.prototype._layout = function(files, cb, extend) {
                 }
             });
 
+            // Create the SVG setter
             Object.defineProperty(this.data.shapes[index], '_svg', {
```
Comment (XhmikosR) [inline lib/svg-sprite/mode/standalone.js:93]: Instead we have this here.

File: lib/svg-sprite/mode/standalone.js (hunk 1 above)
Comment (Kreeg) [reply lib/svg-sprite/mode/standalone.js:100]: Maybe that was a part of the plan to have only setters here? I'm a little confused

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:100]: I was getting an ESLint error (https://eslint.org/docs/rules/accessor-pairs) but happy to disable this one case if we are sure we only need a setter.

Feel free to push to this branch.

Comment (Kreeg) [reply lib/svg-sprite/mode/standalone.js:100]: It's just more likely that originally it just was forgotten. So in this case it is ok.

**Review by Kreeg — APPROVED** [review]

**Comment by XhmikosR** [comment]
I think I'm going to hold off on this for now since we currently don't have tests for all modes.

**Comment by XhmikosR** [comment]
@Kreeg this is now covered but I notice the if conditions are not covered and weren't covered before either. Not sure if something can be improved here.

File: lib/svg-sprite/mode/standalone.js [hunk 3]
```diff
@@ -89,13 +89,22 @@ SVGSpriteStandalone.prototype._layout = function(files, cb, extend) {
...
                 enumerable: false,
                 writable: true
             });
 
-            this.data.shapes[index].__defineSetter__('svg', function(svg) {
-                this._svg = svg;
+            // Create the SVG getter/setter
+            Object.defineProperty(this.data.shapes[index], 'svg', {
+                enumerable: true,
```
Comment (XhmikosR) [inline lib/svg-sprite/mode/standalone.js:100]: Need a double check for these 2 properties

Comment (Kreeg) [reply lib/svg-sprite/mode/standalone.js:100]: it better to either remove this props or set the default [values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty#description) (false)

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:100]: They were needed IIRC.

Comment (Kreeg) [reply lib/svg-sprite/mode/standalone.js:100]: No, they has default values if not provided. It's not critical here but it adds new functionality here. I'd rather go without it

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:100]: Like I said, they are needed:

```
test/svg-sprite/minimal-configuration/modes/defs.test.js (5.687 s)
  ● svg-sprite: default: «defs» mode › creates a visually correct stylesheet resource in CSS format

    TypeError: Cannot redefine property: svg
        at Function.defineProperty (<anonymous>)

      47 |
      48 |         // Create the SVG getter
    > 49 |         Object.defineProperty(dataShape, 'svg', {
         |                ^
      50 |             get() {
      51 |                 return this._svg || shape.getSVG(true, shapeDOM => {
      52 |                     shapeDOM.setAttribute('id', shape.id);
```

Comment (Kreeg) [reply lib/svg-sprite/mode/standalone.js:100]: Hmm, strange. But OK

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:100]: You can try dropping any of the two, maybe I missed something.

Comment (XhmikosR) [reply lib/svg-sprite/mode/standalone.js:100]: @Kreeg ^^

**Pull request merged**